src/components/Scatterplot/Layout.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AxisPlacement from '../AxisPlacement';
+import { axisPlacementType } from '../../utils/proptypes';
+
+const propTypes = {
+  chart: PropTypes.node.isRequired,
+  xAxis: PropTypes.node.isRequired,
+  yAxis: PropTypes.node.isRequired,
+  yAxisPlacement: axisPlacementType,
+};
+
+const defaultProps = {
+  yAxisPlacement: AxisPlacement.RIGHT,
+  contextChart: null,
+};
+
+const axisContainer = area => (axis, placement) => (
+  <div
+    key={area}
+    className="y-axis-container"
+    style={{
+      gridArea: area,
+      display: 'flex',
+      flexDirection: 'column',
+    }}
+  >
+    {React.cloneElement(axis, { yAxisPlacement: placement })}
+  </div>
+);
+
+const Layout = ({ chart, xAxis, yAxis, yAxisPlacement }) => {
+  const yAxes = [];
+  let areas;
+
+  const quote = s => `'${s}'`;
+
+  switch (yAxisPlacement) {
+    case AxisPlacement.LEFT:
+      areas = [
+        // formatting for readability
+        'yaxis chart',
+        '. xaxis',
+        '. context',
+      ]
+        .map(quote)
+        .join(' ');
+      yAxes.push(axisContainer('yaxis')(yAxis, yAxisPlacement));
+      break;
+    case AxisPlacement.BOTH: {
```

Comment:
would be super neat to have xAxis on top placement as well

Reply:
ha, I was going to leave that for a future PR, but I'll go ahead and do it in here.

Reply:
Actually, no -- I'm going to do a separate PR.